### ISSUE TITLE
All examples in documentation need to have e2e testing

### ISSUE THREAD

**Issue by frankfarzan** [kptdev/kpt#1655]
This includes examples in the book and in the reference docs.

**Comment by mikebz** [comment]
I believe this is already done.  @etefera can you comment?  This was supposed to be a part of the Q1 OKRs.

**Comment by etefera** [comment]
Yes, https://raw.githubusercontent.com/GoogleContainerTools/kpt/next/site/guides/consumer/get/README.md is an example of where we're validating the shell commands we're presenting. Is there something missing in that functionality?

**Comment by frankfarzan** [comment]
This includes testing all packages in `package-examples` directory. Currently we have multiple methods for testing:

1. Tests embedded in documentation.
2. Hydration-only: test harness using `.expected` directory
3. Live e2e
4. Packaging e2e?

We need to decide how to get coverage here. 2) is an easy low-hanging fruit.

**Comment by droot** [comment]
@etefera second part in the above list is a `good first issue` to ramp up on the codebase, so assigning it you. Let's have a chat about how this needs to be done.
